pluggy/hooks.py

```diff
@@ -202,28 +202,22 @@ def __init__(self, name, hook_execute, specmodule_or_class=None, spec_opts=None)
         self._wrappers = []
         self._nonwrappers = []
         self._hookexec = hook_execute
-        self._specmodule_or_class = None
         self.argnames = None
         self.kwargnames = None
         self.multicall = _multicall
-        self.spec_opts = spec_opts or {}
+        self.spec = None
         if specmodule_or_class is not None:
+            assert spec_opts is not None
             self.set_specification(specmodule_or_class, spec_opts)
 
     def has_spec(self):
-        return self._specmodule_or_class is not None
+        return True if self.spec is not None else False
```

Comment:
Why not just ``return self.spec is not None``? Much easier to read than the trinary operator.

Reply:
Heh, indeed!